src/publicodes-state/providers/simulationProvider/context.ts

```diff
@@ -33,15 +33,15 @@ type SimulationContextType = {
   everyMosaicChildrenWithParent: Record<DottedName, DottedName[]>
   rawMissingVariables: MissingVariables
   categories: DottedName[]
-  subcategories: Record<DottedName, DottedName[]>
+  subcategories: DottedName[]
   addToEngineSituation: (situationToAdd: Situation) => Situation
   isInitialized: boolean
 }
 export const SimulationContext = createContext<SimulationContextType>({
   rules: undefined,
   engine: undefined,
   pristineEngine: null,
-  safeGetRule: () => null,
+  safeGetRule: () => undefined,
```

Comment:
If `undefined` is the only return type, better to use `void`

Reply:
Never used this, can you show me what you have in mind ?